### PR TITLE
Ignore pipelines without config

### LINKED ISSUE #2312: Don't fail on missing CI config
### Component

server

### Describe the bug

Not necessarily all branches of a repository are intended to have a CI config. But branches without a CI config will always create an error in the UI. 

image

In the past, I was able to work around this issue by adding `[skip ci]` to all commits to branches without a CI config. This is now also broken, see https://github.com/woodpecker-ci/woodpecker/pull/2216.

My suggestion would be to simply ignore missing CI configurations, or at least not create a failed pipeline in the UI. Alternatively, users should have the option to ignore branches. A common example would be `github-pages` branches, where only page artifacts are published and no CI configuration.

### System Info

```shell
{"source":"https://github.com/woodpecker-ci/woodpecker","version":"next-06d1f3c92e"}
```

### Additional context

_No response_

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/woodpecker-ci/woodpecker/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't [already an issue](https://github.com/woodpecker-ci/woodpecker/issues) that reports the same bug to avoid creating a duplicate.
- [X] Checked that the bug isn't fixed in the `next` version already [https://woodpecker-ci.org/faq#which-version-of-woodpecker-should-i-use]
- [X] Check that this is a concrete bug. For Q&A join our [Discord Chat Server](https://discord.gg/fcMQqSMXJy) or the [Matrix room](https://matrix.to/#/#woodpecker:matrix.org).

### PR DESCRIPTION
**Pull request by anbraten**
closes #2312

### CODE REVIEW

**Comment by anbraten** [comment]
Haven't tested it yet

**Comment by 6543** [comment]
yes the forge implementations have to report if its realy a 404 or something else :D

File: server/forge/configFetcher.go [hunk 1]
```diff
@@ -29,6 +29,14 @@ import (
 	"go.woodpecker-ci.org/woodpecker/v2/shared/constant"
 )
 
+type ErrConfigNotFound struct {
+	configs []string
+}
+
+func (m *ErrConfigNotFound) Error() string {
```
Comment (6543) [inline server/forge/configFetcher.go:36]: should implement errors.Is interface

File: server/forge/configFetcher.go [hunk 2]
```diff
@@ -29,6 +29,14 @@ import (
 	"go.woodpecker-ci.org/woodpecker/v2/shared/constant"
 )
 
+type ErrConfigNotFound struct {
```
Comment (6543) [inline server/forge/configFetcher.go:32]: move error to `server/forge/types/errors.go`

File: server/forge/configFetcher.go [hunk 3]
```diff
@@ -172,18 +175,18 @@ func (cf *configFetcher) getFirstAvailableConfig(c context.Context, configs []st
...
 		if fileMeta, found := cf.checkPipelineFile(c, fileOrFolder); found {
-			log.Trace().Msgf("ConfigFetch[%s]: found %s file: '%s'", cf.repo.FullName, userDefinedLog, fileOrFolder)
+			log.Trace().Msgf("ConfigFetch[%s]: found file: '%s'", cf.repo.FullName, fileOrFolder)
 			return fileMeta, nil
 		}
 	}
 
 	// nothing found
-	return nil, fmt.Errorf("%s configs not found searched: %s", userDefinedLog, strings.Join(configs, ", "))
+	return nil, &ErrConfigNotFound{configs: configs}
```
Comment (6543) [inline server/forge/configFetcher.go:191]: the forge drivers should report back if it's a 404 or something else

**Review by 6543 — CHANGES_REQUESTED** [review]

**Review by 6543 — APPROVED** [review]

**Comment by 6543** [comment]
request another review as changed by me

**Comment by 6543** [comment]
@anbraten you might also code comment if you found something

File: server/pipeline/create.go [hunk 4]
```diff
@@ -16,14 +16,16 @@ package pipeline
...
 import (
 	"context"
+	"errors"
 	"fmt"
 	"regexp"
 
 	"github.com/rs/zerolog/log"
 
-	"go.woodpecker-ci.org/woodpecker/v2/pipeline/errors"
+	pipelineErrors "go.woodpecker-ci.org/woodpecker/v2/pipeline/errors"
```
Comment (xoxys) [inline server/pipeline/create.go:25]: We use already `pipeline_errors` in multiple other locations? Would you prefer `pipelineErrors`?

File: server/pipeline/create.go [hunk 5]
```diff
@@ -16,14 +16,16 @@ package pipeline
...
 	"go.woodpecker-ci.org/woodpecker/v2/server"
 	"go.woodpecker-ci.org/woodpecker/v2/server/forge"
+	forgeTypes "go.woodpecker-ci.org/woodpecker/v2/server/forge/types"
```
Comment (xoxys) [inline server/pipeline/create.go:28]: Same here, we use `forge_types` already.

**Review by xoxys — APPROVED** [review]

**Pull request merged**